androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/surveydetail/MultiChoiceForm.kt

```diff
@@ -0,0 +1,102 @@
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.unit.dp
+import co.nimblehq.avishek.phong.kmmic.android.R
+import co.nimblehq.avishek.phong.kmmic.presentation.uimodel.SurveyAnswerUiModel
+
+@Composable
+fun MultiChoiceForm(
+    surveyAnswerUiModels: List<SurveyAnswerUiModel>,
+    modifier: Modifier = Modifier,
+) {
+    var checkedAnswers by remember { mutableStateOf(emptySet<SurveyAnswerUiModel>()) }
+    LazyColumn(modifier = modifier.padding(horizontal = 80.dp)) {
+        items(surveyAnswerUiModels.size) { index ->
+            val answerUiModel = surveyAnswerUiModels[index]
+            val isChecked = checkedAnswers.contains(answerUiModel)
+
+            Choice(
+                answerUiModel = answerUiModel,
+                isChecked = isChecked,
+                onClick = {
+                    checkedAnswers = if (checkedAnswers.contains(answerUiModel)) {
+                        checkedAnswers.minus(answerUiModel)
+                    } else {
+                        checkedAnswers.plus(answerUiModel)
+                    }
+                }
+            )
+
+            if (index < surveyAnswerUiModels.lastIndex) {
+                Divider(color = White, thickness = 1.dp)
+            }
+        }
+    }
+}
+
+@Composable
+private fun Choice(
+    answerUiModel: SurveyAnswerUiModel,
+    isChecked: Boolean,
+    onClick: () -> Unit,
+) {
+    Row(
+        verticalAlignment = Alignment.CenterVertically,
+        modifier = Modifier
+            .fillMaxWidth()
+            .wrapContentHeight()
+            .clickable(onClick = onClick)
+    ) {
+        Text(
+            text = answerUiModel.text.orEmpty(),
+            color = if (isChecked) White else White.copy(alpha = 0.5f),
+            style = if (isChecked) {
+                MaterialTheme.typography.h6
+            } else {
+                MaterialTheme.typography.h6.copy(fontWeight = FontWeight.Normal)
+            },
+            maxLines = 1,
+            overflow = TextOverflow.Ellipsis,
+            modifier = Modifier
+                .weight(1f)
+                .padding(vertical = 14.dp)
+        )
+
+        Image(
+            painter = painterResource(
+                if (isChecked) {
+                    R.drawable.ic_checkbox_checked
+                } else {
+                    R.drawable.ic_checkbox_unchecked
+                }
+            ),
+            contentDescription = null,
+            contentScale = ContentScale.FillWidth
+        )
+    }
+}
+
+@Preview
+@Composable
+fun MultipleChoiceFormPreview(
+    @PreviewParameter(SurveyDetailScreenPreviewParameterProvider::class)
+    params: SurveyDetailScreenPreviewParameterProvider.Params,
+) {
+    MultiChoiceForm(
+        surveyAnswerUiModels = params.survey.questionUiModels[0].answers
```

Comment:
This too shows a lot of unnecessary Previews. Please refactor a bit to reduce the number of previews for this composable.

Reply:
I'll update all the previews of this PR and others onward. 🙇🏽

Reply:
Fixed: https://github.com/nimblehq/avishek-phong-kmm-ic/pull/125/commits/c5c80968d0715422a83b2ba7adce5fc21a12ad7d